### PR TITLE
ref(ui) Remove well from shared-components

### PR DESCRIPTION
**Pull request by markstory**
Make a new react component for well that covers `.well`. I've also added a component for 'hero' sized icons as there were a few icons that relied on being inside a .well to become big. I threw in a bit of button refactoring as well because I was close by and it was easy to do.

If folks are ok with these new components I'll add storybooks for them.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/versionHoverCard.jsx [hunk 1]
```diff
@@ -115,16 +117,16 @@ const VersionHoverCard = createReactClass({
     let {orgId} = this.props;
     return {
       body: (
-        <div className="version-hovercard blankslate m-a-0 p-x-1 p-y-1 align-center">
+        <Box p={2} className="align-center">
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/versionHoverCard.jsx]: This changes the padding by a couple pixels, but increases cohesion with the react-grid layout.

File: src/sentry/static/sentry/app/components/heroIcon.jsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import InlineSvg from 'app/components/inlineSvg';
+
+const HeroIcon = styled(props => {
+  let {size = 72, ...otherProps} = props;
+  return <InlineSvg size={size} {...otherProps} />;
+})`
+  color: ${p => p.theme.gray6};
+`;
+export default HeroIcon;
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/heroIcon.jsx]: If folks have a better name/place for this let me know.

File: src/sentry/static/sentry/app/components/heroIcon.jsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import styled from 'react-emotion';
+
+import InlineSvg from 'app/components/inlineSvg';
+
+const HeroIcon = styled(props => {
+  let {size = 72, ...otherProps} = props;
+  return <InlineSvg size={size} {...otherProps} />;
+})`
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/heroIcon.jsx]: nit: but I would write this

```
const HeroIcon = styled(props => (
  <InlineSvg {...props} />;
)`

HeroIcon.defaultProps = {
  size: '72px'
}
```

Also to note, [`size` needs to be a string](https://github.com/getsentry/sentry/blob/d94c94a6769b8a186f2e1b924e249d4653f56348/src/sentry/static/sentry/app/components/inlineSvg.jsx#L22).

**Review by billyvg — COMMENTED** [review]
Looks good, we will need to check for usage in `getsentry`. I suspect it's used there.

File: src/sentry/static/sentry/app/components/versionHoverCard.jsx (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/versionHoverCard.jsx]: Would you be able to share a screenshot of the diference?

File: src/sentry/static/sentry/less/shared-components.less [hunk 4]
```diff
@@ -2542,54 +2542,6 @@ ul.radio-inputs {
...
-  margin-bottom: 20px;
-  border-radius: 3px;
-
-  p {
-    &:last-child {
-      margin-bottom: 0;
-    }
-  }
-
-  &.important {
```
[the remark this answers is not in the document]
Comment (markstory) [reply src/sentry/static/sentry/less/shared-components.less]: I couldn't find any hits. I haven't checked getsentry though.

File: src/sentry/static/sentry/app/components/versionHoverCard.jsx (hunk 1 above)
Comment (markstory) [reply src/sentry/static/sentry/app/components/versionHoverCard.jsx]: ### Before 
versioncard before

### After

screen shot 2018-09-17 at 10 10 57 am

**Comment by markstory** [comment]
@billyvg @EvanPurkhiser Thank you for the review. I've made the suggested updates.

File: src/sentry/static/sentry/app/components/well.jsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import styled from 'react-emotion';
+import PropTypes from 'prop-types';
+
+const Well = styled('div')`
+  border: 1px solid ${p => p.theme.borderLight};
+  box-shadow: 'none';
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/well.jsx]: Don't need quotes here

File: docs-ui/components/well.stories.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import {storiesOf} from '@storybook/react';
+import {withInfo} from '@storybook/addon-info';
+import {withKnobs, boolean} from '@storybook/addon-knobs';
+
+import Well from 'app/components/well';
+
+const stories = storiesOf('UI|Well', module);
+stories.addDecorator(withKnobs);
```
Comment (billyvg) [inline docs-ui/components/well.stories.js]: Is this needed? I think it's added globally.

**Review by billyvg — COMMENTED** [review]
From looking at the "project release details no commits no deploys" snapshot on percy, I think we'll want a bit more padding at the bottom of the box.

Did you check if there are `well` class names in getsentry?

File: docs-ui/components/well.stories.js (hunk 6 above)
Comment (billyvg) [reply docs-ui/components/well.stories.js]: Hmm doesn't look like it's added globally

**Comment by markstory** [comment]
> Did you check if there are well class names in getsentry? 

There are some hits in getsentry. I'll restore the required selectors in the less file until I can follow up and change getsentry.

File: src/sentry/static/sentry/app/index.js [hunk 7]
```diff
@@ -164,6 +164,7 @@ export default {
     Switch: require('app/components/switch').default,
     GlobalModal: require('app/components/globalModal').default,
     SetupWizard: require('app/components/setupWizard').default,
+    Well: require('app/components/well').default,
```
Comment (markstory) [inline src/sentry/static/sentry/app/index.js]: There are a few places we could use this component in getsentry as well.

**Review by evanpurkhiser — COMMENTED** [review]
Looking at percy it looks like there's some issue with the button in the well (there's a joke in here somewhere)

**Comment by markstory** [comment]
@EvanPurkhiser Very strange, that button renders fine locally in chrome and firefox. I'll try rebuilding.

**Comment by markstory** [comment]
@EvanPurkhiser Visual diffs in percy turned out much better now.

**Review by evanpurkhiser — APPROVED** [review]
🚰 (closest thing I can find to a well)

**Pull request merged**